core-processor/src/ext.rs

```diff
@@ -768,6 +768,19 @@ impl CountersOwner for Ext {
     }
 }
 
+impl Ext {
+    fn atomic_gas<T, E>(&mut self, func: impl FnOnce() -> Result<T, E>) -> Result<T, E> {
+        let amount = self.context.gas_counter;
+
+        func().inspect_err(|_| {
+            self.context.gas_counter = GasCounter::new(amount.burned() + amount.left());
```

Comment:
Is this sum protected against overflow?

Reply:
Good catch! Seems like I have to add a constructor for `GasCounter` then.

Reply:
Resolved in 5a09cf4.